app/controllers/stafftools_controller.rb

```diff
@@ -0,0 +1,30 @@
+class StafftoolsController < ApplicationController
+  layout 'staff'
+
+  before_action :authorize_access
+  before_action :set_resources
+
+  def resources
+  end
+
+  def search
+    respond_to do |format|
+      format.html { render partial: 'stafftools/search_results', locals: { resources: @resources } }
+    end
+  end
+
+  private
+
+  def authorize_access
+    not_found unless current_user.try(:staff?)
+  end
+
+  def set_resources
+    resource_query = params[:query].present? ? match_phrase_prefix(params[:query]) : {}
+    @resources     = StafftoolsIndex::User.query(resource_query).page(params[:page]).per(20)
```

Comment:
Shouldn't this just be `StafftoolsIndex` an not `StafftoolsIndex::User` so you can get organization results as well?